akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/EventStreamClient.scala

```diff
@@ -105,17 +105,19 @@ object EventStreamClient {
   )(implicit ec: ExecutionContext, mat: Materializer): Source[A, NotUsed] = {
 
     def getAndHandleEvents = {
-      def getAndHandle(lastEventId: Option[String]) = {
+      def getAndHandle(lastEventId: Option[String])
+        : (Future[Option[ServerSentEvent]], A) = {
         import EventStreamUnmarshalling._
         val request = {
           val r = Get(uri).addHeader(Accept(`text/event-stream`))
           lastEventId.foldLeft(r)((r, i) => r.addHeader(`Last-Event-ID`(i)))
         }
-        val events = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val events      = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val emptySource = Future.successful(Source.empty[ServerSentEvent])
         Source
-          .fromFuture(events)
+          .fromFuture(events.fallbackTo(emptySource))
```

Comment:
@hseeberger Np, things are looking good, Heiko! Only question is if only failure of the Future is worth parrying for or if you want to also parry against the failure of the returned Source (in which case recoverWithRetries is the way to go instead of fallbackTo) :)

Let me know what you prefer and I'll modify the PR accordingly

√